examples/azure_key/AzureKeyCredential.cs

```diff
@@ -0,0 +1,137 @@
+using Azure.Core;
+using Azure.Identity;
+using Azure.Security.KeyVault.Keys;
+using Azure.Security.KeyVault.Keys.Cryptography;
+using System;
+using System.Buffers.Binary;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using AzureSignatureAlgorithm = Azure.Security.KeyVault.Keys.Cryptography.SignatureAlgorithm;
+
+namespace Tmds.Ssh.AzureKeyExample;
+
+sealed class AzureKeyCredential : PrivateKeyCredential
+{
+    public AzureKeyCredential(string vaultName, string keyName) :
+        base((c) => GetAzureKey(vaultName, keyName, c), $"Azure Key Vault: {vaultName}/{keyName}")
+    { }
+
+    public static async ValueTask<string> GetAzurePubKey(string vaultName, string keyName)
+    {
+        (var _, KeyVaultKey key) = await GetKeyVaultKey(vaultName, keyName);
+
+        if (key.KeyType == KeyType.Rsa)
+        {
+            RSAParameters pubParams = key.Key.ToRSA(includePrivateParameters: false)
+                .ExportParameters(false);
+            return GetRsaPubKey(pubParams);
+        }
+        else if (key.KeyType == KeyType.Ec)
+        {
+            ECParameters pubParams = key.Key.ToECDsa(includePrivateParameters: false)
+                .ExportParameters(false);
+            return GetEcdsaPubKey(pubParams);
+        }
+        else
+        {
+            throw new NotImplementedException($"Unsupported Azure key type {key.KeyType}");
+        }
+    }
+
+    private static async ValueTask<Key> GetAzureKey(string vaultName, string keyName, CancellationToken cancellationToken = default)
+    {
+        (TokenCredential cred, KeyVaultKey key) = await GetKeyVaultKey(vaultName, keyName);
+
+        if (key.KeyType == KeyType.Rsa)
+        {
+            CryptographyClient azureClient = new CryptographyClient(key.Id, cred);
+            RSAParameters pubParams = key.Key.ToRSA(includePrivateParameters: false)
+                .ExportParameters(false);
+
+            return new Key(new AzureRsaKey(azureClient, pubParams));
```

Comment:
side-note: it was intentional that the derived class would be able to return keys of different types as you are doing here.

Reply:
No change required. This is nicely demonstrating how different key types can be loaded by a derived class!

Reply:
Fantastic, just wanted to confirm!